sklearn/tests/test_random_projection.py

```diff
@@ -377,6 +378,51 @@ def test_random_projection_feature_names_out(random_projection_cls):
     assert_array_equal(names_out, expected_names_out)
 
 
+@pytest.mark.parametrize("random_state", range(50))
+@pytest.mark.parametrize("n_rows", (2, 9, 10, 11, 1000))
+@pytest.mark.parametrize("n_cols", (2, 9, 10, 11, 1000))
+@pytest.mark.parametrize("random_projection_cls", all_RandomProjection)
+def test_inverse_transform(random_state, n_rows, n_cols, random_projection_cls):
+    n_components = 10
+    for data in make_sparse_random_data(
+        n_rows, n_cols, n_rows * n_cols // 100 + 1, random_state
+    ):
+        random_projection = random_projection_cls(
+            n_components=n_components, fit_inverse_transform=True
+        )
+        if n_cols < n_components:
+            with pytest.warns(
+                DataDimensionalityWarning,
+                match="The number of components is higher than the number of features",
+            ):
+                projected = random_projection.fit_transform(data)
+        else:
+            projected = random_projection.fit_transform(data)
```

Comment:
I think it would be clearer to filter the warning
```py
with warnings.catch_warnings():
    warnings.filterwarnings(
        "ignore",
        message="The number of components is higher than the number of features",
        category=DataDimensionalityWarning)
    projected = random_projection.fit_transform(data)
```

Reply:
Ok, I'll change that.